arch/arm/core/aarch32/Kconfig

```diff
@@ -40,6 +40,12 @@ config CPU_AARCH32_CORTEX_R
 	help
 	  This option signifies the use of a CPU of the Cortex-R family.
 
+config ARM_ZIMAGE_HEADER
+	bool "zImage Header"
+	depends on CPU_AARCH32_CORTEX_R
```

Comment:
I don't think `ARM_ZIMAGE_HEADER` should depends on `CPU_AARCH32_CORTEX_R`. The `ARM_ZIMAGE_HEADER` is a common feature, so it should be something like A CPU or A SOC or A board select this feature, what do you think?

Reply:
@ayankuma Any feedback here, since there is a code freeze coming up soon?

Reply:
@povergoing @microbuilder Thanks for the review and reminder, I will fix this and send a new version in few hours.

Reply:
I have pushed a new patch and fixed this.